### PR TITLE
NFDIV-633

### PR DESCRIPTION
**Pull request by velimd**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-633

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            argThat(allOf(
+                hasEntry(APPLICATION_REFERENCE, "1234-5678-9012-3456"),
+                hasEntry(SERVICE, "divorce"),
+                hasEntry(PARTNER, "husband"),
+                hasEntry(PAPERS, "divorce papers"),
+                hasEntry(MARRIAGE_CERTIFICATE, "yes"),
+                hasEntry(MARRIAGE_FOREIGN_UNION_CERTIFICATE, "no"),
+                hasEntry(MARRIAGE_FOREIGN_UNION_CERTIFICATE_TRANSLATION, "no"),
+                hasEntry(NAME_CHANGE_PROOF, "yes")
+            )), // NOSONAR
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotificationTest.java:113]: can remove this as you have a suppression

File: src/main/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotification.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    templateVars.put(
+                        caseData.getDivorceOrDissolution().isDivorce()
+                            ? MARRIAGE_FOREIGN_UNION_CERTIFICATE_TRANSLATION
+                            : CIVIL_PARTNERSHIP_FOREIGN_UNION_CERTIFICATE_TRANSLATION, "yes");
+                    break;
+                case NAME_CHANGE_PROOF:
+                    templateVars.put(NotificationConstants.NAME_CHANGE_PROOF, "yes");
+                    break;
+                default:
+                    break;
```
[the remark this answers is not in the document]
Comment (velimd) [reply src/main/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationOutstandingActionNotification.java:110]: You have to set the variable regardless hence i do it before and only change it to yes if the document is there. The default was only set to please the style checker.

**Review by nitinprabhu — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/PaymentMade.java [hunk 3]
```diff
@@ -52,7 +59,13 @@ public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuild
             log.info("Case {} submitted", details.getId());
             data.setDateSubmitted(LocalDateTime.now());
 
-            notification.send(details.getData(), details.getId());
+            CaseData caseData = details.getData();
+            notification.send(caseData, details.getId());
+            if (caseData.getPetitionerWantsToHavePapersServedAnotherWay() == YesOrNo.YES
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/PaymentMade.java:64]: I think this should be implemented as a state validation. So something like

```
State state = Draft;
if (Submitted.validate().isEmpty()) {
  state = Submitted;
} else if (AwaitingDocuments.validate().isEmpty()) {
  state = AwaitingDocuments;
}

and then you update Submitted.validate to check for the documents.

Comment (velimd) [reply src/main/java/uk/gov/hmcts/divorce/citizen/event/PaymentMade.java]: actually, i think it can be cleaner.

**Review by linusnorton — APPROVED** [review]

**Pull request merged**